tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -407,6 +407,48 @@ Status FragmentConsolidator::consolidate_fragments(
         std::to_string(fragment_uris.size()) + " required fragments.");
   }
 
+  // In case we have a dense Array check that the fragments can be consolidated
+  // without data loss
```

Comment:
We should add a little more explanation here.

Reply:
Done! Took inspiration from our docs..